MeshKernelNET/test/MeshKernelNETCoreTest/Api/MeshKernelCurvilinearTest.cs

```diff
@@ -0,0 +1,884 @@
+
+                    Assert.AreEqual(4, curvilinearGrid.NumM);
+
+                    curvilinearGrid.Dispose();
+                    disposableGeometryList.Dispose();
+
+                }
+                finally
+                {
+                    api.DeallocateState(id);
+                }
+            }
+        }
+
+
+        [Test]
+        public void CurvilinearComputeTransfiniteFromSplinesThroughAPI()
+        {
+            // Setup
+            using (var api = new MeshKernelApi())
+            using (var geometryListIn = new DisposableGeometryList())
+            {
+                var id = 0;
+                try
+                {
+                    id = api.AllocateState(0);
+                    
+                    var geometrySeparator = api.GetSeparator();
+                    geometryListIn.GeometrySeparator = geometrySeparator;
+
+                    geometryListIn.XCoordinates = new[]
+                    {
+                            1.340015E+02, 3.642529E+02, 6.927549E+02, geometrySeparator,
+                            2.585022E+02, 4.550035E+02, 8.337558E+02, geometrySeparator,
+                            1.002513E+02, 4.610035E+02, geometrySeparator,
+                            6.522547E+02, 7.197551E+02
+                        };
+
+                    geometryListIn.YCoordinates = new[]
+                    {
+                            2.546282E+02, 4.586302E+02, 5.441311E+02, geometrySeparator,
+                            6.862631E+01, 2.726284E+02, 3.753794E+02, geometrySeparator,
+                            4.068797E+02, 7.912642E+01, geometrySeparator,
+                            6.026317E+02, 2.681283E+02
+                        };
+
+                    geometryListIn.Values = new[]
+                    {
+                            0.0, 0.0, 0.0, geometrySeparator,
+                            0.0, 0.0, 0.0, geometrySeparator,
+                            0.0, 0.0, geometrySeparator,
+                            0.0, 0.0
+                        };
+
+                    geometryListIn.NumberOfCoordinates = geometryListIn.XCoordinates.Length;
+                    var curvilinearParameters = new CurvilinearParameters();
+                    curvilinearParameters.MRefinement = 10;
+                    curvilinearParameters.NRefinement = 10;
+                    curvilinearParameters.SmoothingIterations = 10;
+                    curvilinearParameters.SmoothingParameter = 0.5;
+                    curvilinearParameters.AttractionParameter = 0.0;
+                    Assert.IsTrue(api.CurvilinearComputeTransfiniteFromSplines(id, geometryListIn, curvilinearParameters));
+
+                    var curvilinearGrid = new DisposableCurvilinearGrid();
+                    var success = api.CurvilinearGridGetData(id, out curvilinearGrid);
+                    Assert.IsTrue(success);
+                    curvilinearGrid.Dispose();
+                    geometryListIn.Dispose();
+
```

Comment:
Move the Dispose calls to the finally clause. If one of the Assertions fail this will not be called.